### PR TITLE
refactor(telemetry-utlis): Make MockLogger internal

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

Removes the `MockLogger` class from the `@alpha+@legacy` API surface. It has been deprecated for external use since 2024-06-18. 

## Breaking Changes

- `MockLogger` is now removed from the `@alpha+@legacy` API surface, becoming only `@internal`.
 
## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

[AB#22886](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/22886)

### CODE REVIEW

File: .changeset/green-readers-burn.md
Comment (krauders) [inline .changeset/green-readers-burn.md:1]: Just curious, where does the filename `green-readers-burn.md` come from? Is it randomly generated and these release notes all just get aggregated at release time? Or does the name have some meaning?

**Review by krauders — APPROVED** [review]

File: .changeset/green-readers-burn.md
Comment (alexvy86) [reply .changeset/green-readers-burn.md:1]: Randomly generated. Running `pnpm changeset` from the root of the repo starts a wizard that takes care of that, and just prompts for the package(s) with changes, the "category" of change (like "legacy" in this case; technically free-form, but the ones we use are configured [here](https://github.com/microsoft/FluidFramework/blob/main/fluidBuild.config.cjs#L631-L640) and a value outside that list will cause the release note generation process to complain), a title, and a longer description. You could copy-paste an existing changeset file and give it any name, it doesn't matter in the end as long as they're unique.

Then we have a `flub generate releaseNotes` ([src](https://github.com/microsoft/FluidFramework/blob/main/build-tools/packages/build-cli/src/commands/generate/releaseNotes.ts)) command that takes the changesets and produces the compiled release notes. It runs continuously to produce a preview of the upcoming release notes [here](https://github.com/microsoft/FluidFramework/issues/22101) (always in that issue).

**Comment by alexvy86** [comment]
Still have to check for uses of this class in partner repos before merging, I'll merge once I'm comfortable about that.

**Review by Josmithr — APPROVED** [review]
Thanks!!

**Pull request merged**
